### ISSUE TITLE
Storage upload/download commands provide default values that shoud instead defer to the sdk

### ISSUE THREAD

**Issue by williexu** [Azure/azure-cli#5396]
---
In the storage upload and download commands, we have default values set in the cli.
Ex: `max_connections`  [here](https://github.com/Azure/azure-cli/blob/78163f6b27acaf77dc10030271cd0c10fcfa57e5/src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/operations/blob.py#L169)

We should instead, defer this default value to the sdk.

### Environment summary

Install Method dev_setup.py

**Comment by tjprescott** [comment]
Then why don't we?

**Comment by williexu** [comment]
Because we currently pass arguments to the sdk by wrapping them all up in a dictionary and giving them as kwargs. If an argument is `None`, it will override the default value in the sdk to be `None` as well.

To solve this issue, we would need to add checks in the methods for those args, and only pass them along if they are not `None`. This would need to be done for each method, of which there are several, and for every parameter that has a default value in the sdk, not just `max_connections`. 

All in all, I'll probably do this, I raised this issue for tracking as it is not the point of the [pr](https://github.com/Azure/azure-cli/pull/5384) where I saw this.
